src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java

```diff
@@ -0,0 +1,131 @@
+import org.apache.sling.testing.mock.sling.junit.SlingContext;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Test the creation of paths with specific node types */
+public class MixinsTest {
+
+    @Rule
+    public final SlingContext context = new SlingContext(ResourceResolverType.JCR_OAK);
+    
+    private TestUtil U;
+
+    @Before
+    public void setup() throws RepositoryException, IOException {
+        U = new TestUtil(context);
+    }
+
+    @Test
+    public void addOneMixinOnOnePath() throws Exception {
+        U.parseAndExecute("create path /addOneMixinOnOnePath(nt:unstructured)");
+        U.assertNodeExists("/addOneMixinOnOnePath", "nt:unstructured", Collections.emptyList());
+
+        U.parseAndExecute("add mixin mix:lockable to /addOneMixinOnOnePath");
+        U.assertNodeExists("/addOneMixinOnOnePath", "nt:unstructured", Collections.singletonList("mix:lockable"));
+    }
+
+    @Test
+    public void addTwoMixinsOnTwoPaths() throws Exception {
+        U.parseAndExecute("create path /addTwoMixinsOnOnePath1(nt:unstructured)\n"
+                + "create path /addTwoMixinsOnOnePath2(nt:unstructured)");
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Collections.emptyList());
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Collections.emptyList());
+
+        U.parseAndExecute("add mixin mix:lockable,mix:referenceable to /addTwoMixinsOnOnePath1,/addTwoMixinsOnOnePath2");
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+    }
+
+    @Test
+    public void removeOneMixinFromOnePath() throws Exception {
+        U.parseAndExecute("create path /removeOneMixinOnOnePath(nt:unstructured mixin mix:lockable)");
+        U.assertNodeExists("/removeOneMixinOnOnePath", "nt:unstructured", Collections.singletonList("mix:lockable"));
+
+        U.parseAndExecute("remove mixin mix:lockable from /removeOneMixinOnOnePath");
+        U.assertNodeExists("/removeOneMixinOnOnePath", "nt:unstructured", Collections.emptyList());
+    }
+
+    @Test
+    public void removeTwoMixinsFromTwoPaths() throws Exception {
+        U.parseAndExecute("create path /removeTwoMixinsOnOnePath1(nt:unstructured mixin mix:lockable,mix:referenceable)\n"
+                + "create path /removeTwoMixinsOnOnePath2(nt:unstructured mixin mix:lockable,mix:referenceable)");
+        U.assertNodeExists("/removeTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+        U.assertNodeExists("/removeTwoMixinsOnOnePath2", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+
+        U.parseAndExecute("remove mixin mix:lockable,mix:referenceable from /removeTwoMixinsOnOnePath1,/removeTwoMixinsOnOnePath2");
```

Comment:
Same comment as above about not testing selective removal of mixins

Reply:
Thanks for the review.  I've incorporated the suggested change at: https://github.com/apache/sling-org-apache-sling-jcr-repoinit/pull/38